exonum/src/api/with.rs

```diff
@@ -42,40 +42,135 @@ pub type FutureResult<I> = Box<dyn Future<Item = I, Error = error::Error>>;
 pub struct With<Q, I, R, F> {
     /// Underlying API handler.
     pub handler: F,
+    /// Endpoint actuality.
+    pub actuality: Actuality,
     _query_type: PhantomData<Q>,
     _item_type: PhantomData<I>,
     _result_type: PhantomData<R>,
 }
 
-/// Immutable endpoint marker, which enables creating an immutable kind of `NamedWith`.
-#[derive(Debug)]
-pub struct Immutable;
+/// Endpoint actuality.
+#[derive(Debug, Clone)]
+pub enum Actuality {
+    /// Endpoint is suitable for use.
+    Actual,
+    /// Endpoint is not recommended to use, the support of it will end soon.
+    /// Contains optional value denoting the endpoint expiration date.
+    Deprecated(Option<chrono::Date<chrono::Utc>>),
```

Comment:
Nit: Fully qualified names look weird, IMO. Also, are you sure `Date` (rather than `DateTime`) will always be enough? The `Deprecation` HTTP header draft uses a `DateTime` equivalent, for example.